Delay external imports until we're ready to test tensorboard

pietern opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26033 Skip TestHub on macOS
* **#25993 Delay external imports until we're ready to test tensorboard**
* #25942 Skip TestAutograd.test_deep_reentrant on macOS
* #25930 Refactor macOS build and test
* #25988 Change brew update logic to run much faster
* #25987 Remove trailing whitespace in CircleCI configuration files

These imports fail the test suite if they're not installed, even if we
don't end up testing tensorboard.

[test macos]

Differential Revision: [D17318588](https://our.internmc.facebook.com/intern/diff/D17318588)

pietern committed:
Delay external imports until we're ready to test tensorboard

These imports fail the test suite if they're not installed, even if we
don't end up testing tensorboard.

[test macos]

pietern commented:
These imports (at least the protobuf one) are not available for the macOS test. I don't want to go down the rabbithole of conda/pip protobuf mixing with built-in protobuf and simply moving these imports should do the trick as well.

Fingers crossed this is the final fix needed to re-enable the macOS tests.

orionr reviewed (approved):
Looks reasonable - thanks!

pietern committed:
Update on "Delay external imports until we're ready to test tensorboard"


These imports fail the test suite if they're not installed, even if we
don't end up testing tensorboard.

[test macos]

pietern committed:
Update on "Delay external imports until we're ready to test tensorboard"


These imports fail the test suite if they're not installed, even if we
don't end up testing tensorboard.

[test macos]

Differential Revision: [D17318588](https://our.internmc.facebook.com/intern/diff/D17318588)

pietern committed:
Update on "Delay external imports until we're ready to test tensorboard"


These imports fail the test suite if they're not installed, even if we
don't end up testing tensorboard.

[test macos]

Differential Revision: [D17318588](https://our.internmc.facebook.com/intern/diff/D17318588)
